packages/x-scheduler-headless/src/utils/SchedulerStore/SchedulerStore.utils.ts

```diff
@@ -28,3 +38,189 @@ export function shouldUpdateOccurrencePlaceholder(
 
   return false;
 }
+
+export const DEFAULT_EVENT_MODEL_STRUCTURE: SchedulerEventModelStructure<any> = {};
+
+const EVENT_PROPERTIES_LOOKUP: { [P in keyof CalendarEvent]-?: true } = {
+  id: true,
+  title: true,
+  description: true,
+  start: true,
+  end: true,
+  resource: true,
+  rrule: true,
+  allDay: true,
+  readOnly: true,
+  extractedFromId: true,
+  exDates: true,
+};
+
+const EVENT_PROPERTIES = Object.keys(EVENT_PROPERTIES_LOOKUP) as (keyof CalendarEvent)[];
+
+const RESOURCE_PROPERTIES_LOOKUP: { [P in keyof CalendarResource]-?: true } = {
+  id: true,
+  title: true,
+  eventColor: true,
+};
+
+const RESOURCE_PROPERTIES = Object.keys(RESOURCE_PROPERTIES_LOOKUP) as (keyof CalendarResource)[];
+
+/**
+ * Converts an event model to a processed event using the provided model structure.
+ */
+export function getProcessedEventFromModel<TEvent extends object>(
+  event: TEvent,
+  eventModelStructure: SchedulerEventModelStructure<TEvent> | undefined,
+): CalendarEvent {
+  const processedEvent = {} as CalendarEvent;
+
+  for (const key of EVENT_PROPERTIES) {
+    const getter = eventModelStructure?.[key]?.getter;
+
+    // @ts-ignore
+    processedEvent[key] = getter ? getter(event) : event[key];
+  }
+
+  return processedEvent;
+}
+
+/**
+ * Updates an event model based on the provided changes and model structure.
+ */
+export function getUpdatedEventModelFromChanges<TEvent extends object>(
+  event: TEvent,
+  changes: Partial<CalendarEvent>,
+  eventModelStructure: SchedulerEventModelStructure<TEvent> | undefined,
+): TEvent {
+  let updatedEventModel = { ...event };
+  const propertiesWithSetter: [AnyEventSetter<TEvent>, any][] = [];
+
+  for (const key in changes) {
+    if (changes.hasOwnProperty(key)) {
+      const typedKey = key as keyof CalendarEvent;
+      const setter = eventModelStructure?.[typedKey]?.setter;
+      if (setter) {
+        // @ts-ignore
+        propertiesWithSetter.push([setter, changes[key]]);
+      } else {
+        // @ts-ignore
+        updatedEventModel[key] = changes[key];
+      }
+    }
+  }
+
+  for (const [setter, value] of propertiesWithSetter) {
+    updatedEventModel = setter(updatedEventModel, value);
+  }
+
+  return updatedEventModel;
+}
+
+/**
+ * Create an event model from a processed event using the provided model structure.
+ */
+export function createEventModel<TEvent extends object>(
```

Comment:
Nit: the logic in `getUpdatedEventModelFromChanges` and `createEventModel` seems pretty similar - we could probably extract some of it. I'm fine leaving it as is too, but I couldn't help but notice 🙈